brave/brave-ios
# Fix #6847, #6735: Integrate Solana Swap

## Summary of Changes
- Integrate Solana Swap on iOS into the existing native swap UI
- Solana Swap uses [Jupiter](jup.ag). A disclaimer and link to Jupiter at the bottom of swap, as well as displaying Brave fee.
- Security/privacy review: https://github.com/brave/security/issues/1189

This pull request fixes #6847,
#6735 

## Submitter Checklist:


## Test Plan:
1. Open native wallet and open Swap
2. Verify 0.85% Brave fee and Jupiter is shown at bottom. Jupiter should be a link that contains a disclaimer, with a 'learn more' button to Jupiter
3. Select some to/from tokens, verify sell/buy price is updated when fetching the price quote
4. Create some swap transactions
5. Verify they're displayed in transaction summary / transaction details in Account Details for your Solana account


## Screenshots:

https://user-images.githubusercontent.com/5314553/217845280-60262f90-22fc-4a5a-93cf-2727c609dd44.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
